src/platform_impl/windows/mod.rs

```diff
@@ -413,16 +413,27 @@ impl Menu {
     }
 
     pub fn show_context_menu_for_hwnd(&mut self, hwnd: isize, position: Option<Position>) {
-        let hpopupmenu = self.hpopupmenu;
         unsafe {
             SetWindowSubclass(
                 hwnd as _,
                 Some(menu_subclass_proc),
                 CONTEXT_MENU_SUBCLASS_ID,
-                Box::into_raw(Box::new(self)) as _,
+                dwrefdata_from_obj(self),
             );
         }
-        show_context_menu(hwnd as _, hpopupmenu, position)
+        let rc = show_context_menu(hwnd as _, self.hpopupmenu, position);
+        unsafe {
+            RemoveWindowSubclass(
+                hwnd as _,
+                Some(menu_subclass_proc),
+                CONTEXT_MENU_SUBCLASS_ID,
+            );
+        }
```

Comment:
Not needed, we can remove this and `SetWindowSubclass` before calling `show_context_menu`.